synapse/lib/view.py

```diff
@@ -899,6 +899,13 @@ async def addTrigger(self, tdef):
         tdef.setdefault('async', False)
         tdef.setdefault('enabled', True)
 
+        # Several values may be present in tdef from packed triggers.
+        # Remove them since they are not valid to persist.
+        tdef.pop('startcount', None)
```

Comment:
If we want the API to be resilient against extra keys, rather than creating a denylist we should start with a new empty dict and copy in the values from tdef that are valid.